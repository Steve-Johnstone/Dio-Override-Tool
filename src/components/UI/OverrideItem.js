import { formatOverride } from '../../helpers/formatOverride';
import PropTypes from 'prop-types';

const OverrideItem = ({ override, selectedOverrides, handleClick }) => {
	const overrideClass = selectedOverrides.includes(override)
		? 'override-selected'
		: 'override-list-item';

	return (
		<li onClick={() => handleClick(override)}>
			<span className={overrideClass}>{formatOverride(override)}</span>
		</li>
	);
};

OverrideItem.propTypes = {
	override: PropTypes.string,
	selectedOverrides: PropTypes.array,
	handleClick: PropTypes.func,
};

export default OverrideItem;
