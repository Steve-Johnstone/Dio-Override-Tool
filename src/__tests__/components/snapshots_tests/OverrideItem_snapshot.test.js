import renderer from 'react-test-renderer';
import OverrideItem from '../../../components/UI/OverrideItem';

describe('OverrideItem component', () => {
	const handleClick = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<OverrideItem
					override='homeaway'
					selectedOverrides={['multi-room', 'homeaway']}
					handleClick={handleClick}
				/>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
