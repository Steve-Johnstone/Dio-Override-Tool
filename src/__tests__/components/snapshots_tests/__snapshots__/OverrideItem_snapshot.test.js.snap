// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverrideItem component should display the DOM correctly 1`] = `
<li
  onClick={[Function]}
>
  <span
    className="override-selected"
  >
    Homeaway
  </span>
</li>
`;
